updater/spec/dependabot/integration_spec.rb

```diff
@@ -209,8 +209,12 @@
     context "when there is an exception that blocks PR creation (cloud)" do
       before do
         allow(api_client).to receive(:create_pull_request).and_raise(StandardError, "oh no!")
-        allow(Dependabot::Experiments).to receive(:enabled?).with(:record_ecosystem_versions).and_return(true)
```

Comment:
As far as I can tell doing it this way really only worked while we had a single call against `enabled?`. Since we already had the reset! method I figured we could make use of it instead of mocking things so that later levels of abstraction have an easier time handling things.